### ISSUE TITLE
Support Literal Interference from Exponentiation

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#8434]
Similar to https://github.com/microsoft/pyright/issues/8419, it would also be helpful to support exponentiation:

```python
from typing import reveal_type, Final

MY_CONST: Final = 2**2
 
reveal_type(MY_CONST) # revealed type is `int`
```

We can also end up with `Any` once we reach `2^26` for whatever reason (separate from this, I feel this could lead to subtle bugs):

```python
LARGE_NUM: Final = 2**32 - 1

reveal_type(LARGE_NUM) # revealed type is `Any`.
```

[Playground Link
](https://pyright-play.net/?strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMogCmAboQIYA2A%2BvAoQDRQBiqlAsAFCcCyAmlQGEA8gDkAygBUAXM1YUoAXigAmAFSrlnKJyKlKNRIQAUfQaMkBKKAGICJchUIATWIcwBnKAANUML504AGQBBACUAcQBRKhEAVW4ZFhRKRRV1AGZlKABaKABGAI5dBwM6IxCI6LjuK1tiymdXOg9vYJQ4LwA6QoqomPiqZUS5VLUNADZC%2BupaY16qgeULTiA)

MyPy handles both cases correctly, revealing `Literal[4]`, and `Literal[4294967295]`, so support in Pyright would be nice. At least for me, it's not uncommon to keep constants created through exponentiation, especially when dealing with byte units.

**Comment by erictraut** [comment]
The rules for exponentiation are a bit subtle in python, so I had good reason not to include this operator in pyright's initial implementation of "literal math". I think there are ways we could make this work within some limitations.

I'll note that mypy appears to have a bug where it doesn't apply appropriate limitations and crashes. `MY_CONST: Final = 2**100000`. I'll file a bug on this issue.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by max-muoto** [comment]
> This will be included in the next release.

Thanks Eric, super helpful

**Comment by erictraut** [comment]
This is included in pyright 1.1.372.
